File: Selfie/Selfie/AvatarViewController.m
//
//  AvatarViewController.m
//  Camera
//
//  Created by Joe E. on 10/19/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

#import "AvatarViewController.h"
#import <Parse/Parse.h>

@interface AvatarViewController () <UINavigationControllerDelegate, UIImagePickerControllerDelegate>

@end

@implementation AvatarViewController

- (IBAction)avatarButtonPressed:(id)sender {
    UIImagePickerController * imagePicker = [[UIImagePickerController alloc]init];
    imagePicker.sourceType = UIImagePickerControllerSourceTypeCamera;
    imagePicker.delegate = self;
    [self presentViewController:imagePicker animated:true completion:^{

    }];
    
}



- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
    //attempting to set up the image
    
//    if (imagePicker) {

//        UIIm
//        self.avatarImageButton set
//        self.avatarImageButto n setImage:<#(nullable UIImage *)#> forState:<#(UIControlState)#>
    
//    }

}

- (IBAction)registerButtonPressed:(id)sender {
    PFUser * user = [PFUser user];
    user.username = self.username;
    user.password = self.password;
    user.email = self.email;
    
    [user saveInBackground];
    
    UIStoryboard * storyboard = [UIStoryboard storyboardWithName:@"Create" bundle:nil];
    [storyboard instantiateViewControllerWithIdentifier:@"SelfieTableVC"];

}

- (void)viewDidLoad {
    [super viewDidLoad];
 
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];

}


#pragma mark - Navigation


- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {

}


@end

File: Selfie/Selfie/AvatarViewController.swift
//
//  AvatarViewController.swift
//  Selfie
//
//  Created by Joe E. on 11/3/15.
//  Copyright © 2015 Joe E. All rights reserved.
//

import UIKit
import Parse

private let CREATE = "Create"
private let SELFIE_TABLE_VC = "SelfieTableVC"

class AvatarViewController: UIViewController {
    
    //Properties
    var username = String()
    var password = String()
    var email = String()
    var avatarImage = UIImage()
    
    //MARK: - @IBOutlets
    @IBOutlet var avatarImageButton: UIButton!
    @IBOutlet var registerButon: UIButton!
    
    //MARK: - @IBActions
    @IBAction func avatarButtonPressed(sender:AnyObject) {
        let imagePicker = UIImagePickerController()
        imagePicker.sourceType = .Camera
        imagePicker.delegate = self
        presentViewController(imagePicker, animated: true, completion: nil)
        
    }
    
    @IBAction func registerButtonPressed(sender:AnyObject) {
        let user = PFUser()
        user.username = username
        user.password = password
        user.email = email
        
        user.saveInBackground()
        
        let storyboard = UIStoryboard(name: CREATE, bundle: nil).instantiateViewControllerWithIdentifier(SELFIE_TABLE_VC)
        
        
    }
    

    override func viewDidLoad() {
        super.viewDidLoad()

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()

    }

}

extension AvatarViewController : UINavigationControllerDelegate {
    
}

extension AvatarViewController : UIImagePickerControllerDelegate {
    
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : AnyObject]) {
        
    }
    
}